addons/l10n_it_edi/models/res_config_settings.py

```diff
@@ -1,93 +1,55 @@
-    l10n_it_edi_proxy_current_state = fields.Char(compute='_compute_l10n_it_edi_proxy_current_state')
-    l10n_it_edi_register = fields.Boolean(compute='_compute_l10n_it_edi_register', inverse='_set_l10n_it_edi_register_demo_mode')
-    l10n_it_edi_demo_mode = fields.Selection(
-        [('demo', 'Demo'),
-         ('test', 'Test (experimental)'),
-         ('prod', 'Official')],
-        compute='_compute_l10n_it_edi_demo_mode',
-        inverse='_set_l10n_it_edi_register_demo_mode',
-        readonly=False)
+    l10n_it_edi_register = fields.Boolean(
+        compute='_compute_l10n_it_edi_register',
+        inverse='_set_l10n_it_edi_register',
+        readonly=False,
+    )
+    l10n_it_edi_purchase_journal_id = fields.Many2one(
+        related='company_id.l10n_it_edi_purchase_journal_id',
+        readonly=False,
+    )
+    l10n_it_edi_show_purchase_journal_id = fields.Boolean(compute='_compute_l10n_it_edi_show_purchase_journal_id')
 
     def _create_proxy_user(self, company_id, edi_mode):
-        self.env['account_edi_proxy_client.user']._register_proxy_user(company_id, 'l10n_it_edi', edi_mode)
+        return self.env['account_edi_proxy_client.user']._register_proxy_user(company_id, 'l10n_it_edi', edi_mode)
 
-    def button_create_proxy_user(self):
-        self._create_proxy_user(self.company_id, self.l10n_it_edi_demo_mode)
-
-    @api.depends('company_id.account_edi_proxy_client_ids', 'company_id.account_edi_proxy_client_ids.active')
-    def _compute_l10n_it_edi_demo_mode(self):
-        for config in self:
-            edi_user = self.env['account_edi_proxy_client.user'].search([
-                ('company_id', '=', config.company_id.id),
-                ('proxy_type', '=', 'l10n_it_edi'),
-            ], limit=1)
-            config.l10n_it_edi_demo_mode = edi_user.edi_mode or 'demo'
-
-    @api.depends('company_id.account_edi_proxy_client_ids', 'company_id.account_edi_proxy_client_ids.active')
-    def _compute_is_edi_proxy_active(self):
-        for config in self:
-            config.is_edi_proxy_active = config.company_id.account_edi_proxy_client_ids
-
-    @api.depends('company_id.account_edi_proxy_client_ids', 'company_id.account_edi_proxy_client_ids.active')
-    def _compute_l10n_it_edi_proxy_current_state(self):
+    @api.depends('company_id')
+    def _compute_l10n_it_edi_show_purchase_journal_id(self):
         for config in self:
-            proxy_user = config.company_id.account_edi_proxy_client_ids.search([
-                ('company_id', '=', config.company_id.id),
-                ('proxy_type', '=', 'l10n_it_edi'),
-            ], limit=1)
-
-            config.l10n_it_edi_proxy_current_state = 'inactive' if not proxy_user else 'demo' if proxy_user.id_client[:4] == 'demo' else 'active'
+            # Only show the setting when there exists more than 1 purchase journal.
+            total_purchase_journal = self.env['account.journal'].search_count([
```

Comment:
```suggestion
            purchase_journal_count = self.env['account.journal'].search_count([
```
Nitpick, feel free to ignore :laughing: